C error with `byte` in `for`

**V version:** 0.1.24 0d93eeb
**OS:** Linux

**What did you do?**
```v
fn main() {
  for byte in "hello" {
    a := byte
  }
}
```

**What did you expect to see?**
A compilation error, or maybe a compilation success, not sure.
 
**What did you see instead?**
```
warning: bug.v:3:5: `a` declared and not used
/home/r00ster/.cache/v/bug.tmp.c: In function ‘main__main’:
/home/r00ster/.cache/v/bug.tmp.c:3882:1: warning: statement with no effect [-Wunused-value]
 byte a= byte ;
 ^~~~
/home/r00ster/.cache/v/bug...
(Use `v -g` to print the entire error message)

V error: C error. This should never happen.
Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```

--- a/vlib/compiler/for.v
+++ b/vlib/compiler/for.v
@@ -129,6 +129,7 @@ fn (p mut Parser) for_st() {
 	}
 	// `for val in vals`
 	else if p.peek() == .key_in || p.peek() == .left_arrow {
+		p.check_not_reserved()
 		val := p.check_name()
 		p.fspace()
 		//p.check(.key_in)

